packages/generate/src/edgeql-js.ts

```diff
@@ -196,9 +208,26 @@ export async function generateQueryBuilder(params: {
 
   const futureFilePath = path.join(syntaxOutDir, "future.ts");
 
-  if (options.future) {
-    await fs.writeFile(futureFilePath, headerComment + futureFileContent);
+  let content = headerComment;
+
+  if (
+    options.future ||
+    (options.strictTypeNames && options.polymorphismAsDiscriminatedUnions)
+  ) {
+    content += futureFileContent;
+  } else {
+    if (options.strictTypeNames) {
+      content += strictTypeNamesFileContent;
+    }
+    if (options.polymorphismAsDiscriminatedUnions) {
+      content += polymorphismAsDiscriminatedUnionsFileContent;
+    }
+  }
```

Comment:
Probably need to create this content dynamically from future object.
```ts

const future = {
  polymorphismAsDiscriminatedUnions: false,
  strictTypeNames: false,
  ...options.future,
};

content += `export const future = ${JSON.stringify(future, undefined, 2)} as const;
`;

Reply:
yes, will do sth like this, was thinking to do it if we add more flags, but can do it now too